sycl/test-e2e/KernelCompiler/kernel_compiler_sycl_jit.cpp

```diff
@@ -203,18 +203,21 @@ int test_build_and_run() {
       syclex::properties{syclex::build_options{flags}, syclex::save_log{&log},
                          syclex::registered_kernel_names{"ff_templated<int>"}});
 
-  // extern "C" was used, so the name "ff_cp" is not mangled and can be used
-  // directly.
+  // extern "C" was used, so the name "ff_cp" is implicitly known.
   sycl::kernel k = kbExe2.ext_oneapi_get_kernel("ff_cp");
 
-  // The templated function name will have been mangled. Mapping from original
-  // name to mangled is not yet supported. So we cannot yet do this:
-  // sycl::kernel k2 = kbExe2.ext_oneapi_get_kernel("ff_templated<int>");
+  // The templated function name was registered.
+  sycl::kernel k2 = kbExe2.ext_oneapi_get_kernel("ff_templated<int>");
```

Comment:
so awesome!  This makes me happy.

Reply:
Shout-out to @premanandrao and @jzc who did the heavy lifting for this feature 🙏